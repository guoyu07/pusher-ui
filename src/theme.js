const fontFamilies = [
  '-apple-system',
  'BlinkMacSystemFont',
  'Segoe UI',
  'Roboto',
  'Oxygen',
  'Ubuntu',
  'Cantarell',
  'Fira Sans',
  'Droid Sans',
  'Helvetica Neue',
  'sans-serif',
].map(name => `"${name}"`).join(', ');


export const fontFamily = fontFamilies;
export const white = '#fff';
export const black = '#1e1e1e';
export const grey = '#e7ebf3';
export const darkGrey = '#d6d6d6';
export const darkDarkGrey = '#989898';
export const lightGrey = '#e7ebf3';
export const primary = '#549eff';
export const success = '#319a31';
export const warning = '#ffaf1c';
export const danger = '#ca0808';
export const zIndex1 = 1;
export const zIndex2 = 2;
export default {
  fontFamily,
  white,
  black,
  grey,
  darkGrey,
  darkDarkGrey,
  lightGrey,
  primary,
  success,
  warning,
  danger,
  zIndex1,
  zIndex2,
};
