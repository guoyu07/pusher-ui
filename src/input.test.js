import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'glamorous';

import Input from './input';
import * as theme from './theme';

jest.mock('./label', () => 'Label');

describe('Input', () => {
  const createSnapshot = (props = {}) => {
    expect(
      renderer
        .create(
          <ThemeProvider theme={theme}>
            <Input {...props} />
          </ThemeProvider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  };

  it('should render an input', () => {
    createSnapshot();
  });

  it('should render a disabled input', () => {
    createSnapshot({ disabled: true });
  });

  it('should render a raised input', () => {
    createSnapshot({ raised: true });
  });

  it('should render an input with a validation error', () => {
    createSnapshot({ error: true });
  });

  it('should render inside a label, passing error props if present', () => {
    createSnapshot({ label: 'This is a good input' });

    createSnapshot({ label: 'This is a good input', error: true });
  });

  it('should pass on any css props', () => {
    createSnapshot({ padding: 24, margin: 12, color: 'hotpink' });
  });
});
