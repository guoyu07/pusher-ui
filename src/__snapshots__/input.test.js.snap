// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should pass on any css props 1`] = `
.glamor-0,
[data-glamor-0] {
  position: relative;
  box-sizing: border-box;
  height: 2.5rem;
  padding-top: 0.75rem;
  padding-right: 1.125rem;
  padding-bottom: 0.75rem;
  padding-left: 1.125rem;
  font-family: inherit;
  font-size: 1rem;
  outline: none;
  transition: border-color .18s ease, color .18s ease;
  color: hotpink;
  border: 1px solid transparent;
  border-radius: 2px;
  background-color: #eff4f7;
  padding: 24px;
  margin: 12px;
  -webkit-transition: border-color .18s ease, color .18s ease;
  -moz-transition: border-color .18s ease, color .18s ease;
}

.glamor-0::-webkit-input-placeholder,
[data-glamor-0]::-webkit-input-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::-moz-placeholder,
[data-glamor-0]::-moz-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::-ms-input-placeholder,
[data-glamor-0]::-ms-input-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::placeholder,
[data-glamor-0]::placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::selection,
[data-glamor-0]::selection {
  color: #fff;
  background-color: #006eff;
}

.glamor-0:focus,
[data-glamor-0]:focus {
  caret-color: #006eff;
  box-shadow: 0 0 0 2px rgba(0,110,255,0.3);
}

<input
  className="glamor-0"
  value=""
/>
`;

exports[`Input should render a disabled input 1`] = `
.glamor-0,
[data-glamor-0] {
  position: relative;
  box-sizing: border-box;
  height: 2.5rem;
  padding-top: 0.75rem;
  padding-right: 1.125rem;
  padding-bottom: 0.75rem;
  padding-left: 1.125rem;
  font-family: inherit;
  font-size: 1rem;
  outline: none;
  transition: border-color .18s ease, color .18s ease;
  color: #2b303b;
  border: 1px solid transparent;
  border-radius: 2px;
  background-color: #eff4f7;
  -webkit-transition: border-color .18s ease, color .18s ease;
  -moz-transition: border-color .18s ease, color .18s ease;
}

.glamor-0::-webkit-input-placeholder,
[data-glamor-0]::-webkit-input-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::-moz-placeholder,
[data-glamor-0]::-moz-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::-ms-input-placeholder,
[data-glamor-0]::-ms-input-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::placeholder,
[data-glamor-0]::placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::selection,
[data-glamor-0]::selection {
  color: #fff;
  background-color: #006eff;
}

.glamor-0:focus,
[data-glamor-0]:focus {
  caret-color: #006eff;
  box-shadow: 0 0 0 2px rgba(0,110,255,0.3);
}

<input
  className="glamor-0"
  disabled={true}
  value=""
/>
`;

exports[`Input should render a raised input 1`] = `
.glamor-0,
[data-glamor-0] {
  position: relative;
  box-sizing: border-box;
  height: 2.5rem;
  padding-top: 0.75rem;
  padding-right: 1.125rem;
  padding-bottom: 0.75rem;
  padding-left: 1.125rem;
  font-family: inherit;
  font-size: 1rem;
  outline: none;
  transition: border-color .18s ease, color .18s ease;
  color: #2b303b;
  border: none;
  border-radius: 2px;
  box-shadow: 0 1px 2px 0 rgba(50, 52, 58, 0.15);
  -webkit-transition: border-color .18s ease, color .18s ease;
  -moz-transition: border-color .18s ease, color .18s ease;
}

.glamor-0::-webkit-input-placeholder,
[data-glamor-0]::-webkit-input-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::-moz-placeholder,
[data-glamor-0]::-moz-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::-ms-input-placeholder,
[data-glamor-0]::-ms-input-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::placeholder,
[data-glamor-0]::placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::selection,
[data-glamor-0]::selection {
  color: #fff;
  background-color: #006eff;
}

.glamor-0:focus,
[data-glamor-0]:focus {
  caret-color: #006eff;
  box-shadow: 0 0 0 2px rgba(0,110,255,0.3);
}

<input
  className="glamor-0"
  value=""
/>
`;

exports[`Input should render an input 1`] = `
.glamor-0,
[data-glamor-0] {
  position: relative;
  box-sizing: border-box;
  height: 2.5rem;
  padding-top: 0.75rem;
  padding-right: 1.125rem;
  padding-bottom: 0.75rem;
  padding-left: 1.125rem;
  font-family: inherit;
  font-size: 1rem;
  outline: none;
  transition: border-color .18s ease, color .18s ease;
  color: #2b303b;
  border: 1px solid transparent;
  border-radius: 2px;
  background-color: #eff4f7;
  -webkit-transition: border-color .18s ease, color .18s ease;
  -moz-transition: border-color .18s ease, color .18s ease;
}

.glamor-0::-webkit-input-placeholder,
[data-glamor-0]::-webkit-input-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::-moz-placeholder,
[data-glamor-0]::-moz-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::-ms-input-placeholder,
[data-glamor-0]::-ms-input-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::placeholder,
[data-glamor-0]::placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::selection,
[data-glamor-0]::selection {
  color: #fff;
  background-color: #006eff;
}

.glamor-0:focus,
[data-glamor-0]:focus {
  caret-color: #006eff;
  box-shadow: 0 0 0 2px rgba(0,110,255,0.3);
}

<input
  className="glamor-0"
  value=""
/>
`;

exports[`Input should render an input with a validation error 1`] = `
.glamor-0,
[data-glamor-0] {
  position: relative;
  box-sizing: border-box;
  height: 2.5rem;
  padding-top: 0.75rem;
  padding-right: 1.125rem;
  padding-bottom: 0.75rem;
  padding-left: 1.125rem;
  font-family: inherit;
  font-size: 1rem;
  outline: none;
  transition: border-color .18s ease, color .18s ease;
  color: #ff455c;
  border: 1px solid #ff455c;
  border-radius: 2px;
  background-color: #eff4f7;
  -webkit-transition: border-color .18s ease, color .18s ease;
  -moz-transition: border-color .18s ease, color .18s ease;
}

.glamor-0::-webkit-input-placeholder,
[data-glamor-0]::-webkit-input-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::-moz-placeholder,
[data-glamor-0]::-moz-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::-ms-input-placeholder,
[data-glamor-0]::-ms-input-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::placeholder,
[data-glamor-0]::placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::selection,
[data-glamor-0]::selection {
  color: #fff;
  background-color: #ff455c;
}

.glamor-0:focus,
[data-glamor-0]:focus {
  caret-color: #ff455c;
  box-shadow: 0 0 0 2px rgba(255,69,92,0.3);
}

<input
  className="glamor-0"
  value=""
/>
`;

exports[`Input should render inside a label, passing error props if present 1`] = `
.glamor-0,
[data-glamor-0] {
  position: relative;
  box-sizing: border-box;
  height: 2.5rem;
  padding-top: 0.75rem;
  padding-right: 1.125rem;
  padding-bottom: 0.75rem;
  padding-left: 1.125rem;
  font-family: inherit;
  font-size: 1rem;
  outline: none;
  transition: border-color .18s ease, color .18s ease;
  color: #2b303b;
  border: 1px solid transparent;
  border-radius: 2px;
  background-color: #eff4f7;
  -webkit-transition: border-color .18s ease, color .18s ease;
  -moz-transition: border-color .18s ease, color .18s ease;
}

.glamor-0::-webkit-input-placeholder,
[data-glamor-0]::-webkit-input-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::-moz-placeholder,
[data-glamor-0]::-moz-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::-ms-input-placeholder,
[data-glamor-0]::-ms-input-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::placeholder,
[data-glamor-0]::placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::selection,
[data-glamor-0]::selection {
  color: #fff;
  background-color: #006eff;
}

.glamor-0:focus,
[data-glamor-0]:focus {
  caret-color: #006eff;
  box-shadow: 0 0 0 2px rgba(0,110,255,0.3);
}

<Label
  error={false}
>
  This is a good input
  <input
    className="glamor-0"
    value=""
  />
</Label>
`;

exports[`Input should render inside a label, passing error props if present 2`] = `
.glamor-0,
[data-glamor-0] {
  position: relative;
  box-sizing: border-box;
  height: 2.5rem;
  padding-top: 0.75rem;
  padding-right: 1.125rem;
  padding-bottom: 0.75rem;
  padding-left: 1.125rem;
  font-family: inherit;
  font-size: 1rem;
  outline: none;
  transition: border-color .18s ease, color .18s ease;
  color: #ff455c;
  border: 1px solid #ff455c;
  border-radius: 2px;
  background-color: #eff4f7;
  -webkit-transition: border-color .18s ease, color .18s ease;
  -moz-transition: border-color .18s ease, color .18s ease;
}

.glamor-0::-webkit-input-placeholder,
[data-glamor-0]::-webkit-input-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::-moz-placeholder,
[data-glamor-0]::-moz-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::-ms-input-placeholder,
[data-glamor-0]::-ms-input-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::placeholder,
[data-glamor-0]::placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::selection,
[data-glamor-0]::selection {
  color: #fff;
  background-color: #ff455c;
}

.glamor-0:focus,
[data-glamor-0]:focus {
  caret-color: #ff455c;
  box-shadow: 0 0 0 2px rgba(255,69,92,0.3);
}

<Label
  error={true}
>
  This is a good input
  <input
    className="glamor-0"
    value=""
  />
</Label>
`;

exports[`Input should take a value prop 1`] = `
.glamor-0,
[data-glamor-0] {
  position: relative;
  box-sizing: border-box;
  height: 2.5rem;
  padding-top: 0.75rem;
  padding-right: 1.125rem;
  padding-bottom: 0.75rem;
  padding-left: 1.125rem;
  font-family: inherit;
  font-size: 1rem;
  outline: none;
  transition: border-color .18s ease, color .18s ease;
  color: #2b303b;
  border: 1px solid transparent;
  border-radius: 2px;
  background-color: #eff4f7;
  -webkit-transition: border-color .18s ease, color .18s ease;
  -moz-transition: border-color .18s ease, color .18s ease;
}

.glamor-0::-webkit-input-placeholder,
[data-glamor-0]::-webkit-input-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::-moz-placeholder,
[data-glamor-0]::-moz-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::-ms-input-placeholder,
[data-glamor-0]::-ms-input-placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::placeholder,
[data-glamor-0]::placeholder {
  font-family: inherit;
  color: #95979d;
}

.glamor-0::selection,
[data-glamor-0]::selection {
  color: #fff;
  background-color: #006eff;
}

.glamor-0:focus,
[data-glamor-0]:focus {
  caret-color: #006eff;
  box-shadow: 0 0 0 2px rgba(0,110,255,0.3);
}

<input
  className="glamor-0"
  value="hello!"
/>
`;
